src/components/Map.jsx

```diff
@@ -1,46 +1,140 @@
 import Map from '@arcgis/core/Map';
 import esriConfig from '@arcgis/core/config';
+import { whenOnce } from '@arcgis/core/core/reactiveUtils';
+import Polygon from '@arcgis/core/geometry/Polygon';
+import { union } from '@arcgis/core/geometry/geometryEngine';
+import FeatureLayer from '@arcgis/core/layers/FeatureLayer';
 import MapView from '@arcgis/core/views/MapView';
 import LayerSelector from '@ugrc/layer-selector';
 import '@ugrc/layer-selector/src/LayerSelector.css';
 import { useEffect, useRef, useState } from 'react';
+import { fieldNames } from '../../functions/common/config';
+import { useSearchMachine } from '../SearchMachineProvider';
+import stateOfUtah from '../data/state-of-utah.json';
+
+const stateOfUtahPolygon = new Polygon(stateOfUtah);
+const stateOfUtahExtent = stateOfUtahPolygon.extent;
 
 esriConfig.assetsPath = '/assets';
 
-const initialExtent = {
-  type: 'extent',
-  xmax: -11762120.612131765,
-  xmin: -13074391.513731329,
-  ymax: 5225035.106177688,
-  ymin: 4373832.359194187,
-  spatialReference: 3857,
-};
 export default function MapComponent() {
-  const mapDiv = useRef(null);
+  const [state, send] = useSearchMachine();
   const [selectorOptions, setSelectorOptions] = useState(null);
+
+  const map = useRef(null);
+  const view = useRef(null);
+
+  const mapDiv = useRef(null);
   useEffect(() => {
-    const map = new Map();
+    map.current = new Map();
 
-    const view = new MapView({
+    view.current = new MapView({
       container: mapDiv.current,
-      map,
-      extent: initialExtent,
+      map: map.current,
+      extent: stateOfUtahExtent,
     });
 
     setSelectorOptions({
-      view,
+      view: view.current,
       quadWord: import.meta.env.VITE_DISCOVER_KEY,
-      baseLayers: ['Hybrid', 'Lite', 'Terrain', 'Topo', 'Color IR'],
+      baseLayers: ['Lite', 'Terrain', 'Topo', 'Hybrid', 'Color IR'],
     });
 
     return () => {
-      view.destroy();
-      map.destroy();
+      view.current.destroy();
+      map.current.destroy();
     };
   }, []);
 
+  const searching = useRef(false);
+  useEffect(() => {
+    if (!state.matches('searching')) {
+      return;
+    }
+
+    async function searchLayer(layer) {
+      try {
+        const featureLayer = new FeatureLayer({
+          url: layer[fieldNames.queryLayers.featureService],
+        });
+        map.current.add(featureLayer);
+
+        // does this really make sure that all features have been loaded?
+        const layerView = await view.current.whenLayerView(featureLayer);
```

Comment:
**issue**: i got an error on public water system facilities when adding the layer view.

Reply:
That's because it doesn't yet have a feature service prop. I've changed the schema check to require that field so it should not show up in the app if it's missing.

Reply:
961c357bcf1e35b1821f03ce0f38288ddb0db837